Makefile

```diff
@@ -186,18 +190,26 @@ build: setup go-build
 PILOT_GO_BINS:=${ISTIO_BIN}/pilot-discovery ${ISTIO_BIN}/pilot-agent \
                ${ISTIO_BIN}/istioctl ${ISTIO_BIN}/sidecar-initializer
 $(PILOT_GO_BINS): depend
-	bin/gobuild.sh ${ISTIO_BIN}/$(@F) istio.io/istio/pilot/tools/version ./pilot/cmd/$(@F)
+	bin/gobuild.sh $@ istio.io/istio/pilot/tools/version ./pilot/cmd/$(@F)
+
+# Non-static istioctls. These are typically a build artifact so placed in out/ rather than bin/ .
+# The if/findstring is used to replace osx with darwin, win with windows and anything else (currently linux) with the suffix after the dash.
+# Cross-compiling for Mac on a Linux system seems to have fewer issues if CGO_ENABLED=0 (this is now handled by gobuild.sh).
+ISTIOCTL_LIST:=istioctl-linux istioctl-osx istioctl-win.exe
+$(foreach TGT,$(ISTIOCTL_LIST),$(eval ${OUT}/$(TGT): depend; \
```

Comment:
Not sure about this - I think we want a cross-compile to be able to build all binaries, not only istioctl, and should 
be something at top level. 

Also please make sure a normal build doesn't do all cross-compilations.

Reply:
A normal build will create "istioctl", and if/when the functionality for defaulting to native GOOS is added then this binary will match the OS of the local system.  The change I've added follows the behavior of the existing upload-istioctl script and creates the different/unique names of istioctl-linux, istioctl-osx, and istioctl-win.exe that don't overlap with the existing istioctl.  Nothing currently depends on the new targets I'm adding so I wouldn't expect any kind of build to build them unless unless explicitly requested (unless I've overlooked some place that does a dynamic lookup of all the targets exposed in the makefile).

The other approach I can do is wait for or add cross-compilation support and then have the build run "make istioctl" three times (once for each OS).  I'm hesitant to go down this approach since it's quite a kludge and would make it harder to eventually have (if desired) a single make target that does all the end-to-end steps for a release build.